### PR TITLE
migrate from databaseUsername to databaseAccount and fully use MariaDBAccount

### PR DESCRIPTION
**Pull request by zzzeek**

### CODE REVIEW

**Comment by zzzeek** [comment]
for whatever reason when i was reviewing this I totally missed that there are *two* databases, so those had to be fully spec'ed out separately.    the volumes were changed to secrets, which is good, but that means we had to move from configmaps to secrets.

File: controllers/octavia_controller.go [hunk 1]
```diff
@@ -803,7 +900,7 @@ func (r *OctaviaReconciler) apiDeploymentCreateOrUpdate(instance *octaviav1.Octa
 		deployment.Spec = instance.Spec.OctaviaAPI
 		deployment.Spec.DatabaseInstance = instance.Spec.DatabaseInstance
 		deployment.Spec.DatabaseHostname = instance.Status.DatabaseHostname
-		deployment.Spec.DatabaseUser = instance.Spec.DatabaseUser
+		deployment.Spec.DatabaseAccount = instance.Spec.DatabaseAccount
```
Comment (zzzeek) [inline controllers/octavia_controller.go:903]: need to add the other DB here also

File: controllers/octavia_controller.go [hunk 2]
```diff
@@ -703,19 +643,155 @@ func (r *OctaviaReconciler) reconcileNormal(ctx context.Context, instance *octav
 		instance.Status.Conditions.MarkTrue(amphoraControllerReadyCondition(octaviav1.Worker), condition.DeploymentReadyMessage)
 	}
 
+	// remove finalizers from unused MariaDBAccount records
+	err = mariadbv1.DeleteUnusedMariaDBAccountFinalizers(ctx, helper, octavia.DatabaseCRName, instance.Spec.DatabaseAccount, instance.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = mariadbv1.DeleteUnusedMariaDBAccountFinalizers(ctx, helper, octavia.PersistenceDatabaseCRName, instance.Spec.DatabaseAccount, instance.Namespace)
```
Comment (zzzeek) [inline controllers/octavia_controller.go:652]: wrong account name

**Comment by zzzeek** [comment]
OK if it passes it should be good

**Review by beagles — APPROVED** [review]
/lgtm

**Pull request merged**